crates/re_arrow_store/src/store_gc.rs

```diff
@@ -17,4 +23,202 @@ impl std::fmt::Display for GarbageCollectionTarget {
     }
 }
 
-// TODO(#1619): Implement garbage collection.
+impl DataStore {
+    /// Triggers a garbage collection according to the desired `target`.
+    ///
+    /// Returns the list of `RowId`s that were purged from the store.
+    ///
+    /// ## Semantics
+    ///
+    /// Garbage collection works on a row-level basis and is driven by [`RowId`] order, i.e. the order
+    /// defined by the clients' wall-clocks, allowing it to drop data across the different timelines
+    /// in a fair, deterministic manner.
+    /// Similarly, out-of-order data is supported out of the box.
+    ///
+    /// The garbage collector doesn't deallocate data in and of itself: all it does is drop the
+    /// store's internal references to that data (the `DataCell`s), which will be deallocated once
+    /// their reference count reaches 0.
+    ///
+    /// ## Limitations
+    ///
+    /// The garbage collector is currently unaware of our latest-at semantics, i.e. it will drop
+    /// old data even if doing so would impact the results of recent queries.
+    /// See <https://github.com/rerun-io/rerun/issues/1803>.
+    //
+    // TODO(#1804): There shouldn't be any need to return the purged `RowId`s, all secondary
+    // datastructures should be able to purge themselves based solely off of
+    // [`DataStore::oldest_time_per_timeline`].
+    //
+    // TODO(#1803): The GC should be aware of latest-at semantics and make sure they are upheld when
+    // purging data.
+    pub fn gc(&mut self, target: GarbageCollectionTarget) -> (Vec<RowId>, DataStoreStats) {
+        crate::profile_function!();
+
+        self.gc_id += 1;
+
+        let stats_before = DataStoreStats::from_store(self);
+        let initial_nb_rows = stats_before.total_temporal_rows;
+        // NOTE: only temporal data and row metadata get purged!
+        let initial_size_bytes = (stats_before.total_temporal_size_bytes
+            + stats_before.metadata_registry_size_bytes) as f64;
+
+        let row_ids = match target {
+            GarbageCollectionTarget::DropAtLeastPercentage(p) => {
+                assert!((0.0..=1.0).contains(&p));
```

Comment:
It makes no sense to only allow dropping 0-1% of the store

```suggestion
            GarbageCollectionTarget::DropAtLeastFraction(p) => {
                assert!((0.0..=1.0).contains(&p));
```